src/main/java/hudson/plugins/mercurial/MercurialSCMSource.java

```diff
@@ -43,160 +49,265 @@
+    }
+
+    @Override protected void retrieve(@CheckForNull SCMSourceCriteria criteria, @Nonnull SCMHeadObserver observer,
+                            @CheckForNull SCMHeadEvent<?> event, @Nonnull final TaskListener listener)
+            throws IOException, InterruptedException {
+        try (MercurialSCMSourceRequest request= new MercurialSCMSourceContext<>(criteria, observer)
+                .withTraits(traits)
+                .newRequest(this, listener) ) {
+            MercurialInstallation inst = MercurialSCM.findInstallation(request.installation());
+            if (inst == null) {
+                listener.error("No configured Mercurial installation");
+                return;
+            }
+            if (!inst.isUseCaches()) {
+                listener.error("Mercurial installation " + request.installation() + " does not support caches");
+                return;
+            }
+            final Node node = Jenkins.getInstance();
+            if (node == null) { // Should not happen BTW
+                listener.error("Cannot retrieve the Jenkins master node");
+                return;
+            }
+            Launcher launcher = node.createLauncher(listener);
+            StandardUsernameCredentials credentials = getCredentials(request.credentialsId());
+            final FilePath cache = Cache.fromURL(request.source(), credentials, inst.getMasterCacheRoot())
+                    .repositoryCache(inst, node, launcher, listener, true);
+            if (cache == null) {
+                listener.error("Could not use caches, not fetching branch heads");
+                return;
+            }
+            final HgExe hg = new HgExe(inst, credentials, launcher, node, listener, new EnvVars());
+            try {
+                String heads = hg.popen(cache, listener, true,
+                        new ArgumentListBuilder("heads", "--template", "{node} {branch}\\n"));
+                int count = 0;
+                for (String line : heads.split("\r?\n")) {
+                    final String[] nodeBranch = line.split(" ", 2);
+                    final String name = nodeBranch[1];
+                    count++;
+                    if (request.process(new SCMHead(name),
+                            new SCMSourceRequest.RevisionLambda<SCMHead, MercurialRevision>() {
+                                @Override
+                                public @Nonnull
+                                MercurialRevision create(@Nonnull SCMHead branch) {
+                                    return new MercurialRevision(branch, nodeBranch[0]);
+                                }
+                            }, new SCMSourceRequest.ProbeLambda<SCMHead, MercurialRevision>() {
+                                @Override
+                                public @Nonnull
+                                SCMSourceCriteria.Probe create(@Nonnull SCMHead branch, @Nullable final
+                                MercurialRevision revision) {
```

Comment:
Do you have your IDE set to some ridiclously small fill column or something? This method header could all be on one line…